lib/tasks/hyacinth/ci.rake

```diff
@@ -27,66 +23,63 @@ namespace :hyacinth do
       task.fail_on_error = true
     end
 
-  rescue LoadError => e
-    puts "[Warning] Exception creating rspec rake tasks.  This message can be ignored in environments that intentionally do not pull in the RSpec gem (i.e. production)."
-    puts e
-  end
+    desc 'Set rails environment to "test"'
+    task :set_rails_test_environment do
+      ENV['RAILS_ENV'] = 'test'
+      Rails.env = ENV['RAILS_ENV']
+    end
 
-  desc "CI build without rubocop"
-  task :ci_nocop do
-    ENV['RAILS_ENV'] = 'test'
-    Rails.env = ENV['RAILS_ENV']
-    Rake::Task["hyacinth:ci_task"].invoke
-  end
+    desc 'CI build without rubocop'
+    task ci_nocop: ['hyacinth:set_rails_test_environment', 'hyacinth:setup:config_files', 'hyacinth:docker:setup_config_files', :environment, 'hyacinth:ci_specs']
 
-  desc "CI build with Rubocop validation"
-  task ci: ['hyacinth:rubocop'] do
-    ENV['RAILS_ENV'] = 'test'
-    Rails.env = ENV['RAILS_ENV']
-    Rake::Task["hyacinth:ci_task"].invoke
-  end
+    desc 'CI build with Rubocop validation'
+    task ci: ['hyacinth:set_rails_test_environment', 'hyacinth:setup:config_files', 'hyacinth:docker:setup_config_files', :environment, 'hyacinth:rubocop', 'hyacinth:ci_specs']
 
-  desc "CI build"
-  task ci_task: [:environment] do
-    Jettywrapper.jetty_dir = File.join(Rails.root, 'jetty-test')
+    desc 'CI build just running specs'
+    task ci_specs: :environment do
+      rspec_system_exit_failure_exception = nil
 
-    unless File.exists?(Jettywrapper.jetty_dir)
-      puts "\n"
-      puts 'No test jetty found.  Will download / unzip a copy now.'
-      puts "\n"
+      docker_wrapper do
+        duration = Benchmark.realtime do
+          Rake::Task["hyacinth:fedora:reload_cmodels"].invoke
+          Rake::Task["uri_service:db:drop_tables_and_clear_solr"].invoke
+          Rake::Task["hyacinth:test:clear_default_asset_home_content"].invoke
+          Rake::Task["hyacinth:test:clear_default_service_copy_home_content"].invoke
+          Rake::Task["hyacinth:test:clear_access_copy_content"].invoke
+          Rake::Task["uri_service:db:setup"].invoke
+          Rake::Task["db:drop"].invoke
+          Rake::Task["db:create"].invoke
+          Rake::Task["db:migrate"].invoke
+          Rake::Task["db:seed"].invoke
+          ENV['CLEAR'] = 'true' # Set ENV variable for reindex task
+          Rake::Task['hyacinth:index:reindex'].invoke
+          ENV['CLEAR'] = nil  # Clear ENV variable because we're done with it
+          Rake::Task['hyacinth:test:setup_test_project'].invoke
+          Rake::Task['hyacinth:rspec'].invoke
+        end
+        puts "\nCI run finished in #{duration} seconds."
+      end
     end
 
-    Rake::Task["jetty:clean"].invoke
-    Rake::Task["hyacinth:setup:solr_cores"].invoke
+    def docker_wrapper(&block)
```

Comment:
thanks for pointing out this docker_wrapper method. it's a really nice approach.

Reply:
Thanks!  I may bring it over to Hyacinth 3 in the near future too!